libs/agent-sm/bootstrap/src/main/java/org/opensearch/javaagent/bootstrap/AgentPolicy.java

```diff
@@ -28,7 +28,61 @@ public class AgentPolicy {
     private static final Logger LOGGER = Logger.getLogger(AgentPolicy.class.getName());
     private static volatile Policy policy;
     private static volatile Set<String> trustedHosts;
-    private static volatile Set<String> classesThatCanExit;
+    private static volatile BiFunction<Class<?>, List<Class<?>>, Boolean> classesThatCanExit;
+
+    /**
+     * None of the classes is allowed to call {@link System#exit} or {@link Runtime#halt}
+     */
+    public static final class NoneCanExit implements BiFunction<Class<?>, List<Class<?>>, Boolean> {
+        /**
+         * NoneCanExit
```

Comment:
Does the javadoc enforcement require adding these redundant constructors just to add a useless comment?

Reply:
Yep :(